src/core/ui/components/modal/modal.tsx

```diff
@@ -1,16 +1,33 @@
+import { useEffect } from 'react';
 import { useSafeLayoutEffect } from '../../../../shared/hooks';
 import { useAppearance } from '../../context/appearance';
 import { usePopover, useScrollLock } from '../../hooks';
 import { BasicPortal } from '../../portal';
 import { ModalBackdrop } from './backdrop';
 import { ModalContent } from './modal-content';
 
-export const Modal = ({ children }) => {
-  const { floating } = usePopover({
+export const Modal = ({
+  children,
+  handleClose,
+  handleOpen,
+}: React.PropsWithChildren<{
+  handleOpen?: () => void;
+  handleClose?: () => void;
+}>) => {
+  const { floating, isOpen } = usePopover({
     defaultOpen: true,
     autoUpdate: false,
   });
 
+  useEffect(() => {
+    if (!isOpen) {
+      handleClose?.();
+    } else {
+      handleOpen?.();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isOpen]);
```

Comment:
After talking with our design team, we decided to revert the removal of this (close signin modal when clicking outside).
Although this seemed like a good options at first. The sign-in modal is
also used to connect wallets, email and other to your existing account.
There is no way for the user to escape once started this flow, which is
a terrible ux.